presto-native-execution/presto_cpp/main/tests/DynamicFunctionRegistererTest.cpp

```diff
@@ -0,0 +1,223 @@
+      std::string fn_name,
+      std::string nameSpace = "") {
+    facebook::presto::registerPrestoFunction<TestDynamicFunctionMock, bool>(
+        fn_name.c_str(), nameSpace.c_str());
+  }
+  void UdfVarcharDynamicFnRegistration(
+      std::string fn_name,
+      std::string nameSpace = "") {
+    facebook::presto::registerPrestoFunction<
+        TestDynamicFunctionMock2,
+        facebook::velox::Varchar,
+        facebook::velox::Varchar>(fn_name.c_str(), nameSpace.c_str());
+  }
+
+  void writeToFile(const std::string& filePath, std::string_view data) {
+    std::ofstream outputFile(filePath, std::ofstream::binary);
+    outputFile.write(data.data(), data.size());
+    outputFile.close();
+  }
+
+  void emulateConfigFile(std::string_view json, std::string pathStr) {
+    // auto path = exec::test::TempFilePath::create();
+    // auto pathStr = path->getPath();
+    std::filesystem::path fsPath(pathStr);
+    auto fileName = fsPath.filename().string();
+    std::ofstream{fsPath.parent_path() / "test.so"}.put('a');
+
+    // Create dummy shared library to emulate the shared library path
+    // validation.
+    auto filePathForDummySharedFile = exec::test::TempFilePath::create();
+    std::filesystem::path sharedFilePath(filePathForDummySharedFile->getPath());
+    auto dummySharedFile = sharedFilePath.replace_extension(kFileExtension);
+
+    writeToFile(pathStr, json);
+  }
+
+ private:
+  std::shared_ptr<SystemConfig> systemConfig;
+};
+
+TEST_F(DynamicFunctionRegistererTest, twoFunctions) {
+  std::string_view json = R"(
+  {
+    "dynamicLibrariesUdfMap": {
+      "tmp": {
+        "mock1": [
+          {
+            "entrypoint": "registry123",
+            "fileName" : "test",
+            "outputType": "boolean",
+            "paramTypes": [
+            ]
+          }
+        ],
+        "mock2": [
+          {
+            "fileName" : "test",
+            "outputType": "varchar",
+            "paramTypes": [
+              "varchar"
+            ]
+          }
+        ]
+      }
+    }
+  })";
+
+  // Emulate user provided config file.
+  auto path = exec::test::TempFilePath::create();
+  auto pathStr = path->getPath();
+  emulateConfigFile(json, pathStr);
+
+  // auto path = exec::test::TempFilePath::create();
+  // auto pathStr = path->getPath();
+  // std::filesystem::path fsPath(pathStr);
+  // auto fileName = fsPath.filename().string();
+  // std::ofstream{fsPath.parent_path() / "test.so"}.put('a');
+
+  // // Create dummy shared library to emulate the shared library path
+  // validation. auto filePathForDummySharedFile =
+  // exec::test::TempFilePath::create(); std::filesystem::path
+  // sharedFilePath(filePathForDummySharedFile->getPath()); auto dummySharedFile
+  // = sharedFilePath.replace_extension(kFileExtension);
+
+  // writeToFile(pathStr, json);
+
+  // Check functions do not exist first.
+  auto registeredFnSignaturesBefore = velox::getFunctionSignatures();
+  EXPECT_TRUE(
```

Comment:
Write functions for existsBefore/After or notExistsBefore/After and use them in the tests instead of doing a find and checking the returned iterator.